test/image/png_images/test_png_file.py

```diff
@@ -1,39 +1,29 @@
-import os
-
-import pytest
+from pathlib import Path
 
 import fpdf
-from fpdf.errors import FPDFException
-from test.utilities import assert_pdf_equal, relative_path_to
+from test.utilities import assert_pdf_equal
+
+
+HERE = Path(__file__).resolve().parent
 
 
 class TestInsertPNGSuiteFiles:
     def test_insert_png_files(self, tmp_path):
         pdf = fpdf.FPDF(unit="pt")
         pdf.compress = False
 
-        not_supported = [
+        not_supported = {
             "e59ec0cfb8ab64558099543dc19f8378.png",  # Interlacing not supported:
             "6c853ed9dacd5716bc54eb59cec30889.png",  # 16-bit depth not supported:
             "ac6343a98f8edabfcc6e536dd75aacb0.png",  # Interlacing not supported:
             "93e6127b9c4e7a99459c558b81d31bc5.png",  # Interlacing not supported:
             "18f9baf3834980f4b80a3e82ad45be48.png",  # Interlacing not supported:
             "51a4d21670dc8dfa8ffc9e54afd62f5f.png",  # Interlacing not supported:
-        ]
-
-        images = [
-            relative_path_to(f)
-            for f in os.listdir(relative_path_to("."))
-            if f.endswith(".png") and os.path.basename(f) not in not_supported
-        ]
-        images.sort()
-
-        for image in images:
-            if os.path.basename(image) in not_supported:
-                with pytest.raises(FPDFException):
```

Comment:
This logic of testing that an exception was raised has been lost in your refactor

Reply:
Right, good point. Before we had:

```
images = [
    relative_path_to(f)
    for f in os.listdir(relative_path_to("."))
    if f.endswith(".png") and os.path.basename(f) not in not_supported
]
images.sort()

for image in images:
    if os.path.basename(image) in not_supported:
        with pytest.raises(FPDFException):
            pdf.image(x=0, y=0, w=0, h=0, link=None)
```

So there was `os.path.basename(f) not in not_supported` in the first list comprehension which eliminated all files that were in `not_supported`. So did that exception ever happen?

Reply:
Good point! Alright then